### PR TITLE
Cleanup rustdoc-internals

### PR DESCRIPTION
**Pull request by JohnTitor**
- fixed one broken link
- updated date reference
	- removed the mentions to `unindent-comments` pass as per https://github.com/rust-lang/rust/commit/7803a411517f0f1e64d8f37ce377c684d1f2d1fd

cc #1344

### CODE REVIEW

**Comment by JohnTitor** [comment]
cc @rust-lang/rustdoc it'd be great if someone from the team could review this, I'm not familiar with rustdoc :)

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Thanks a lot for the update! :)

Let's check if I have r+ rights here:

@bors r+

**Comment by GuillaumeGomez** [comment]
Ah, it's not handled by bors in here. Well, you got my approval to merge. :)

**Comment by JohnTitor** [comment]
Thanks both! This repo needs one approval from reviewers with write access, so going to cc @rust-lang/wg-rustc-dev-guide :)

**Review by camelid — APPROVED** [review]
Rubber-stamping as rustc-dev-guide team member, not rustdoc team member ;)

**Pull request merged**
